### ISSUE TITLE
tensorflow_ros retrain replaces underscores in labels with spaces

### ISSUE THREAD

**Issue by JannisBush** [tue-robotics/image_recognition#66]
Hey, 

If you label some images using the annotation tool, have underscores in the names of the labels and use `rosrun tensorflow_ros retrain ...`, the underscores will be replaced with spaces in `output_lables.txt`. 
If you now use the `object_recognition_node` to test some new images, you can't just merge the  `unverified` and `verified` folders afterwards to generate a new and larger training set, because the names differ (underscores vs spaces). The spaces in the labels also could lead to some other problems. 

I don't know, if this is the intended behavior (because in your example underscores are used).

Inserting an underscore in line 263 in `tf_retrain.py` will fix this problem.
Old: `label_name = re.sub(r'[^a-z0-9]+', ' ', dir_name.lower())`
New: `label_name = re.sub(r'[^a-z0-9_]+', ' ', dir_name.lower())`

**Comment by MatthijsBurgh** [comment]
I think we manually add the underscores in `output_lables.txt` at the moment. We prefer to have the underscore.

**Comment by LoyVanBeek** [comment]
@MatthijsBurgh leaving the labels unmodified by default would make more sense IMHO.

**Comment by MatthijsBurgh** [comment]
@LoyVanBeek I am absolutely fine with that. But the current code does modify it, because underscores are removed by the script. Which part? I don't know.

**Comment by LoyVanBeek** [comment]
@MatthijsBurgh According to the issue, that happens here https://github.com/tue-robotics/image_recognition/blob/master/tensorflow_ros/src/tensorflow_ros/tf_retrain.py#L263

**Comment by MatthijsBurgh** [comment]
@LoyVanBeek I thought that part would replace the underscore with space. But that is incorrect. I would say to @JannisBush create a PR with the suggested fix.

**Comment by JannisBush** [comment]
Done

**Comment by MatthijsBurgh** [comment]
Fixed by #67
